PiwikTracker/Device.swift

```diff
@@ -0,0 +1,132 @@
+import UIKit
+
+internal struct Device {
+    
+    /// A shared instance holding the properties for the current device
+    static let current:Device = Device(platform: currentPlatform(),
```

Comment:
I would suggest removing the static `current` and have a factory method that returns the current device
````
static func makeCurrentDevice() ->  Device {
  let platform = currentPlatform()
  ....
   return Device(platform: platform, ...)
}
````

I would only create a global shared instance later on if it becomes needed.
Most probably the "tracker class" can have a simple instance variable it can instantiate at construction by calling the static method above (reducing coupling and making testing more simple).